### DISCUSSION TITLE
Count the number of whitespaces with TreeSitter?

### DISCUSSION THREAD

**Discussion by bhargavah** [tree-sitter/tree-sitter#3463] [Q&A] [answered]
I am trying to figure out if a javascript file is minified or not and I was thinking if we could use TreeSitter for that purpose. I mean we parse the file with the help of the TreeSitter parser, get the tokens and count the number of whitespaces etc with the help of TreeSitter api's. Based on the data collected, decide whether the file is minified or not.

Do you think we could achieve this with TreeSitter?

TreeSitter looks much simpler to me but only an expert with the internal details of the TreeSitter would be able to put some light on this. Thank you.

**Comment by amaanq** [comment, marked as answer]
You could totally achieve this with tree-sitter, though it won't exactly be trivial since you'd be building a lot of functionality on top of it. You would first query for nodes that must *not* have whitespace before/after them (likely statements, block start/end, etc, whatever it may be). You could do it by querying the code and just processing each match, or by using a `TreeCursor` yourself to iterate over the tree.

**Reply by bhargavah** [reply]
Thanks @amaanq . I think I am near to what I would like to have. Parse using treesitter and do an DFS over all the token and keep taking a call there.
